build-system/release-workflows/cut-nightly.js

```diff
@@ -48,48 +40,76 @@ async function cutNightlyBranch() {
   );
 
   for (const {sha} of commits.data) {
-    const commitStatus = await octokit.rest.repos.getCombinedStatusForRef({
-      ...params,
-      ref: sha,
-    });
-    const {state} = commitStatus.data;
-
-    log('Status of commit', cyan(sha), 'is', colorizeState(state));
-
-    if (state === 'success') {
-      const response = await octokit.rest.git.updateRef({
+    const {'check_runs': checkRuns} = (
+      await octokit.rest.checks.listForRef({
         ...params,
-        ref: 'heads/nightly',
-        sha,
-      });
+        ref: sha,
+      })
+    ).data;
+    if (checkRuns.some(({status}) => status != 'completed')) {
+      log(
+        'Not all check runs for commit',
+        cyan(sha),
+        'are completed. Checking next commit...'
+      );
+      continue;
+    }
 
-      // Casting to Number because the return type in Octokit is incorrectly
-      // annotated to only ever return 200.
-      switch (Number(response.status)) {
-        case 201:
-          log('Cut a new', cyan('nightly'), 'with', cyan(sha));
-          break;
-        case 200:
-          log(
-            'The',
-            cyan('nightly'),
-            'branch is already at the latest',
-            green('green'),
-            'sha',
-            cyan(sha)
-          );
-          break;
-        default:
-          log(
-            red('An error occurred while attempting to fast-forward the'),
-            cyan('nightly'),
-            red('branch to sha'),
-            cyan(sha)
-          );
-      }
+    if (
+      !checkRuns.every(({conclusion}) =>
+        ['success', 'neutral', 'skipped'].includes(conclusion ?? '')
+      )
+    ) {
+      log(
+        'Not all check runs for commit',
+        cyan(sha),
+        'are successful. Checking next commit...'
+      );
+      continue;
+    }
 
-      break;
+    log(
+      'All check runs for commit',
+      cyan(sha),
+      'have completed successfully. Cutting a new nightly at this commit...'
+    );
+    const response = await octokit.rest.git.updateRef({
+      ...params,
+      ref: 'heads/nightly',
+      sha,
+    });
+
+    // Casting to Number because the return type in Octokit is incorrectly
+    // annotated to only ever return 200.
+    switch (Number(response.status)) {
+      case 201:
+        log(
+          'A new',
+          cyan('nightly'),
+          'branch was successfully cut at commit',
+          cyan(sha)
+        );
+        break;
+      case 200:
+        log(
+          'The',
+          cyan('nightly'),
+          'branch is already at the latest',
+          green('green'),
+          'commit',
+          cyan(sha)
+        );
+        break;
+      default:
+        log(
+          red('An error occurred while attempting to fast-forward the'),
```

Comment:
nit: log error if it isn't logged already

Reply:
Done.